src/scheduler/pbs_sched.c

```diff
@@ -1273,3 +1188,140 @@ main(int argc, char *argv[])
+ *
+ *
+ * @return	int
+ * @retval	1	-	if both command are duplicate
+ * @reval	0	-	if not duplicate
+ *
+ */
+static int
+compare_modify_sched_cmd(sched_cmd *cmd_cache, sched_cmd *new_cmd)
+{
+	if (new_cmd->cmd == SCH_SCHEDULE_RESTART_CYCLE )
+		return 1;
+	
+	if (new_cmd->cmd == SCH_SCHEDULE_AJOB &&
+		(cmd_cache->cmd != SCH_SCHEDULE_AJOB && cmd_cache->cmd != SCH_CONFIGURE &&
+		 cmd_cache->cmd != SCH_RULESET && cmd_cache->cmd != SCH_ERROR)) {
+		return 1;
+	} else if (cmd_cache->cmd == SCH_SCHEDULE_AJOB &&
+		(new_cmd->cmd != SCH_SCHEDULE_AJOB && new_cmd->cmd != SCH_CONFIGURE &&
+		 new_cmd->cmd != SCH_RULESET && new_cmd->cmd != SCH_ERROR)) {
+		cmd_cache->cmd = new_cmd->cmd;
+		cmd_cache->jid = NULL;	
+		return 1;	
+	}
+
+	if (cmd_cache && new_cmd) {
+		if (cmd_cache->jid == NULL && new_cmd->jid == NULL) {
+			if (cmd_cache->cmd == new_cmd->cmd)
+				return 1;
+		} else {
+			if (cmd_cache->cmd == new_cmd->cmd && !strcmp(cmd_cache->jid, new_cmd->jid))
+				return 1;
+		}
+	}
+	return 0;
+}
+
+/**
+ * @brief Filters scheduler commands queue according to the criteria given in the function
+ *              compare_modify_sched_cmd
+ *
+* @param[in] sched_cmds	- scheduler command queue
+ *
+ * @return	ds_queue *
+ * @retval	success	- returns the filtered queue
+ * @reval	 fail -	NULL
+ *
+ */
+ds_queue *
+filter_sched_cmds(ds_queue *sched_cmds)
+{
+	ds_queue *sched_cmds_filtered = new_ds_queue();
+	sched_cmd *tmp_cmd;
+	int cmds_front;
+	int cmds_rear;
+	sched_cmd *cmd_enq;
+
+	if (sched_cmds_filtered == NULL) 
+		goto error;
+
+	cmds_front = sched_cmds->front;
+	cmds_rear = sched_cmds->rear;
+
+	tmp_cmd = sched_cmds->content_arr[cmds_front];
+	if (tmp_cmd == NULL) 
+		goto error;
+
+	cmd_enq = new_sched_cmd();
+	if (cmd_enq == NULL)
+		goto error;
+
+	cmd_enq->cmd = tmp_cmd->cmd;
+	cmd_enq->from_sock = tmp_cmd->from_sock;
+	if (tmp_cmd->jid != NULL) {
+		cmd_enq->jid = malloc(strlen(tmp_cmd->jid) + 1);
+		if (cmd_enq->jid == NULL)
+			goto error;
+		strcpy(cmd_enq->jid, tmp_cmd->jid);
+	}
+
+	if (!ds_enqueue(sched_cmds_filtered, cmd_enq)) 
+		goto error;
+	cmds_front++;
+
+	for (; cmds_front < cmds_rear; cmds_front++) {
+		int no_enq = 0;
+		int cmds_fltrd_front = sched_cmds_filtered->front;
+		int cmds_fltrd_rear = sched_cmds_filtered->rear;
+		for (; cmds_fltrd_front < cmds_fltrd_rear; cmds_fltrd_front++) {
+			if (compare_modify_sched_cmd(sched_cmds_filtered->content_arr[cmds_fltrd_front],
+				sched_cmds->content_arr[cmds_front]) == 1) {
+					no_enq = 1;
+					break;
+			} 
+		}
+		if (no_enq == 0) {
+			sched_cmd *cmd_enq;
+			sched_cmd *tmp_cmd;
+			cmd_enq = new_sched_cmd();
+			if (cmd_enq == NULL)
+				goto error;
+			tmp_cmd = sched_cmds->content_arr[cmds_front];
+
+			cmd_enq->cmd = tmp_cmd->cmd;
+			cmd_enq->from_sock = tmp_cmd->from_sock;
+			if (tmp_cmd->jid != NULL) {
+				cmd_enq->jid = malloc(strlen(tmp_cmd->jid) + 1);
+				if (cmd_enq->jid == NULL)
+					goto error;
+				strcpy(cmd_enq->jid, tmp_cmd->jid);
```

Comment:
here as well, not sure why we are not checking the return value.

Reply:
Please look into my reply given above for the similar comment.